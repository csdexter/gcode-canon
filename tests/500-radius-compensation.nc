(testing radius compensation)
(machine setup)
M05 S600
M09
M23
M49
M69

(turn servos ON)
M17

(control setup)
G15
G17
G23
G40
G49
G50
G69
G90

(in mm)
G21

(we need a tool selected for radius compensation to make sense)
M06 T1 (test output will vary by D of T1)
(initial move to prime buffer)
G01 X10 Y10 F100

(activate left and initial move to set direction)
G41 Y50
(keep going)
X50
Y10
(stub move to clear the corner in a compensated way)
X45
(deactivate and return)
G40 X5 Y5

(initial move)
X10 Y10
(activate right and initial move to set direction)
G42 Y50
(keep going)
X50
Y10
(stub move to clear the corner in a compensated way)
X45
(deactivate and return)
G40 X5 Y5

M2 (THE END)
