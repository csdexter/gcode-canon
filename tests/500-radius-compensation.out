MSG: WAR: Machine servos activated!
MSG: STA: Scanning input for programs (O words)
MSG: WAR: Machine servos activated!
MPOS,10.00,10.00,0.00
MPOS,10.00,50.75,0.00
MPOS,50.75,50.75,0.00
MPOS,50.75,9.25,0.00
MPOS,-45.05,9.25,0.00
MPOS,5.00,5.00,0.00
MPOS,10.00,10.00,0.00
MPOS,10.00,49.25,0.00
MPOS,49.25,49.25,0.00
MPOS,49.25,10.75,0.00
MPOS,-44.95,10.75,0.00
MPOS,5.00,5.00,0.00
